lib/sensu/client/process.rb

```diff
@@ -85,6 +86,9 @@ def publish_check_result(check)
           :client => @settings[:client][:name],
```

Comment:
Should just add `:client_key => @settings[:client][:client_key]`, which will just be `nil` by default. We can then check for `nil?()` when processing.